lms/views/error.py

```diff
@@ -79,3 +85,10 @@ def includeme(config):
         http_server_error, context=httpexceptions.HTTPServerError, **view_defaults
     )
     config.add_exception_view(error, context=Exception, **view_defaults)
+
+    validation_view_defaults = dict(view_defaults)
+    validation_view_defaults["renderer"] = "lms:templates/validation_error.html.jinja2"
+
+    config.add_exception_view(
+        validation_error, context=ValidationError, **validation_view_defaults
```

Comment:
This is why we needed a custom `lms.validation.ValidationError` class. Without it validation failures would raise a standard `pyramid.httpexceptions.HTTPUnprocessableEntity` with additional attributes on it (e.g. the `.messages` attr containing the webargs error messages structure). Webargs errors need to be handled differently than other errors. In particular they need a different error page template to render the error messages. So they need a separate error view. I wouldn't want to register an error view for `HTTPUnprocessableEntity` and assume that its a webargs error and has a webargs `messages` attr because something else could have raised a standard Pyramid `HTTPUnprocessableEntity`. By using a separate exception class for webargs errors, we can easily register a separate error view and template just for them